db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementTablePermission.java

```diff
@@ -220,6 +220,10 @@ protected boolean hasPermissionOnTable(LanguageConnectionContext lcc,
 							Authorizer.PUBLIC_AUTHORIZATION_ID,
 							dbo);
 				}
+				if (rd == null && lcc.getCurrentGroupUser(activation) != null) {
```

Comment:
The current sequence of permission checks in the function hasPermissionOnTable() is the following:
1. whether PUBLIC has permission on this database object
2. if NONE/UNAUTHORIZED, whether the currentUser has permission on this database object
3 if NONE, whether PUBLIC has schema level permission for the schema this database object belongs to.
4. if NONE/UNAUTHORIZED, whether currentUser has schema level permission for the schema this database object belongs to.
5. If NONE, check the role permissions from PUBLIC/currentUser.

With the addition of CurrentGroupUser, In addition to the check for roles from the currentGroupUser, I think we also need to add check
2.5, If NONE, whether the currentGroupUser has permission on this database object
4.5 If NONE, whether currentGroupUser has schema level permission for the schema this database object belongs to.